### PR TITLE
ICU-21967 Remove obsolete references in makedata VS projects

### PR DESCRIPTION
**Pull request by jbrianceau**
These references should have been removed in ICU-20489 and ICU-21420.
Take the opportunity to update the doc as well.

##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21967
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [ ] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [x] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Review by jefgen — APPROVED** [review]
Thank you very much for filing a ticket and cleaning this up.

The changes look good to me.

File: docs/userguide/icu4c/index.md [hunk 1]
```diff
@@ -154,31 +154,28 @@ In the descriptions below, `<ICU>` is the full path name of the ICU4C directory
...
-            plural forms), with its own makefile <b>resfiles.mk</b>.</li>
+            plural forms).</li>
 
             <li><b>lang/</b> Locale data for names of languages, scripts, and locale
-            key names and values, with its own makefile <b>resfiles.mk</b>.</li>
+            key names and values.</li>
 
             <li><b>region/</b> Locale data for names of regions, with its own
-            makefile <b>resfiles.mk</b>.</li>
+            makefile.</li>
```
Comment (daniel-ju) [inline docs/userguide/icu4c/index.md:173]: nit: Can we also remove "with its own makefile" here?

Comment (jbrianceau) [reply docs/userguide/icu4c/index.md]: Indeed 😅 I've just updated the PR accordingly, thanks!

Comment (daniel-ju) [reply docs/userguide/icu4c/index.md]: Thank you! We'll also need to squash the commits into a single commit that has the ticket number in the message, or else our PR checks won't pass.

**Review by daniel-ju — APPROVED** [review]

**Review by jefgen — APPROVED** [review]
Thanks!

**Pull request merged**